Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -33,16 +35,61 @@ struct PurchasedTransactionData {
 
 }
 
+/// The possible results when posting a transaction.
+enum TransactionPosterResult {
+
+    /// Transaction was handled and backend returned a `CustomerInfo`
+    /// or it was computed offline.
+    case success(CustomerInfo)
+
+    /// Transaction had already been posted.
+    case alreadyPosted
```

Comment:
New type helps ensure that we handle this scenario everywhere that used `Result<CustomerInfo, BackendError>`.